src/pages/filter/ui/panels/function-panel/components/compound-het.tsx

```diff
@@ -1,112 +1,130 @@
 import React, { ReactElement, useEffect, useState } from 'react'
 import { Option } from 'react-dropdown'
-import { Form, FormikProps } from 'formik'
+import { observer } from 'mobx-react-lite'
 
 import { IStatFuncData } from '@declarations'
 import { FuncStepTypesEnum } from '@core/enum/func-step-types-enum'
 import filterStore from '@store/filter'
 import { DropDown } from '@ui/dropdown'
+import { DisabledVariantsAmount } from '@pages/filter/ui/query-builder/ui/disabled-variants-amount'
+import {
+  ConditionJoinMode,
+  TFuncCondition,
+  TVariant,
+} from '@service-providers/common/common.interface'
+import { ICompoundHetCachedValues } from '../function-panel.interface'
+import functionPanelStore from '../function-panel.store'
 import { PanelButtons } from './panelButtons'
 
-export interface ICompoundHetFormValues {
-  variants: string[]
-  approx: string | null
-  state?: string | null
-}
-
 const options = [
   { label: 'shared transcript', value: '' },
   { label: 'shared gene', value: 'gene' },
   { label: 'non-intersecting transcripts', value: 'rough' },
 ]
 
-export const CompundHet = ({
-  setFieldValue,
-  values: { approx, variants },
-  submitForm,
-}: FormikProps<ICompoundHetFormValues>): ReactElement => {
-  const cachedValues = filterStore.readFilterCondition<ICompoundHetFormValues>(
-    FuncStepTypesEnum.CompoundHet,
-  )
+export const CompundHet = observer((): ReactElement => {
+  const cachedValues =
+    functionPanelStore.getCachedValues<ICompoundHetCachedValues>(
+      FuncStepTypesEnum.CompoundHet,
+    )
+
+  const { simpleVariants, filterName, filterGroup } = functionPanelStore
 
   const [statFuncStatus, setStatFuncStatus] = useState<IStatFuncData>()
 
-  const initialApprox = cachedValues?.approx || options[0].value
+  const initialApprox: string =
+    cachedValues?.conditions.approx || options[0].value
 
-  const fetchStatFuncAsync = async (
-    param?: Record<string, string | string[]>,
-  ) => {
+  const handleSetConditions = (approx: string | null, variants?: string[]) => {
```

Comment:
It's a good practice to separate functions that are not directly related to the component. Whether move outside the component, or move to an external hook, or move to a store in order to prevent generating this function over and over each time the component updates.